var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -277,6 +277,11 @@ def libs(self):
     # and https://github.com/spack/spack/pull/21408
     patch("bootstrap-toolset.patch", when="@1.75")
 
+    # Allow building context asm sources with GCC on Darwin
+    # See https://github.com/spack/spack/pull/24889
+    # and https://github.com/boostorg/context/issues/177
+    patch("context-macho-gcc.patch")
```

Comment:
Does this patch apply unconditionally to all versions? I would limit it to darwin at the very least to reduce redundant rebuilds for people on linux.

Reply:
I don't know for which versions this applies. The file itself is quite old (2009?), suggesting that it would apply to older Boost versions as well. I have not checked.

Yes, we can restrict to Darwin. Is it also possible to restrict to when GCC is used as compiler?

Reply:
Done.